t/cobrand/fixmystreet.t

```diff
@@ -208,4 +208,120 @@ FixMyStreet::override_config {
     };
 };
 
+for my $test (
+    {
+        update_permission => 'staff',
+        problem_state => 'confirmed',
+    },
+    {
+        update_permission => 'none',
+        problem_state => 'confirmed',
+    },
+    {
+        update_permission => 'none',
+        problem_state => 'closed',
+    },
+    {
+        update_permission => 'staff',
+        problem_state => 'closed',
+    },
+    {
+        update_permission => 'reporter-open',
+        problem_state => 'closed',
+    },
+    {
+        update_permission => 'open',
+        problem_state => 'closed',
+    }
+) {
+    subtest 'Cobrand set to deny updates' => sub {
+        FixMyStreet::override_config {
+            COBRAND_FEATURES => {
+            updates_allowed => { birmingham => $test->{update_permission} },
+            },
+    }, sub {
+        my $problem = FixMyStreet::DB->resultset('Problem')->create(
+        {
+        postcode     => 'EH99 1SP',
+        latitude     => 1,
+        longitude    => 1,
+        areas        => 1,
+        title        => 'to be sent',
+        detail       => 'detail',
+        used_map     => 1,
+        user_id      => 1,
```

Comment:
This is causing the tests to fail as it needs a user. There's a `$mech->create_user_ok` method you can use to set one up and then use the id of the created user here.

Reply:
Although actually it's probably easier to use `$mech->create_problems_for_body` to create the problem as it'll do most of the work.

Reply:
It probably works if you run them in isolation as it only creates the one user which will have Id 1 but when run as part of the suite it cleans out the database between tests which means the auto increment takes effect and the user created at the top of the script is no longer Id 1.